### PR TITLE
Short Data Descriptors

### PR DESCRIPTION
**Pull request by reox**
Hi!
As I found out, it is possible that data descriptors do not have a signature. See https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT 

> 4.3.9.3 Although not originally assigned a signature, the value 
> 0x08074b50 has commonly been adopted as a signature value 
> for the data descriptor record.  Implementers should be 
> aware that ZIP files may be encountered with or without this 
> signature marking data descriptors and SHOULD account for
> either case when reading ZIP files to ensure compatibility.

I added an extra check for that and the template works now with files that have this signature not set.

### CODE REVIEW

File: Templates/ZIPTemplateAdv.bt [hunk 1]
```diff
@@ -513,10 +513,19 @@ typedef struct {
 	if( deFileNameLength > 0 )
 		char     deFileName[ deFileNameLength ];
 	local int len = deExtraFieldLength;
+
 	while (len > 0)
 	{
-		EXTRAFIELD deExtraField;
-		len -= deExtraField.efDataSize + 4;
+        Printf("at --> %d\n", FTell());
```
Comment (strazzere) [inline Templates/ZIPTemplateAdv.bt]: Can we remove this (debug?) statement?

File: Templates/ZIPTemplateAdv.bt [hunk 2]
```diff
@@ -648,14 +653,25 @@ void SeekToDataDescription()
...
     // Fix position
-    FSeek(start+skip-1);
-    ZIPDATADESCR dataDescr;
+    FSeek(start+skip-1-16);
+    if (ReadUInt(FTell()) == S_ZIPDATADESCR){
+        FSeek(start+skip-1-16);
+        ZIPDATADESCR dataDescr;
+    }
+    else{
+        Printf("Found a short header...\n");
```
Comment (strazzere) [inline Templates/ZIPTemplateAdv.bt]: Can we remove this (debug?) statement?

**Comment by strazzere** [comment]
Added a few comments, can we remove the printf/debug statements? After that I'll happily accept all the other changes, thank you!

**Comment by reox** [comment]
Hey, sorry i saw now this is over a year old....

Sure, just remove it.
I also saw that the code is mixed tabs/spaces. What is the code convention here? I could fix that as well then...

ty!

**Comment by strazzere** [comment]
Thanks for taking a look at that @reox :D

**Pull request merged**